sp/src/game/client/mapbase/vscript_vgui.cpp

```diff
@@ -0,0 +1,1871 @@
+
+	for ( int i = 0; i < count; ++i )
+	{
+		g_pScriptVM->GetValue( ax, i, &vx );
+		g_pScriptVM->GetValue( ay, i, &vy );
+
+		px[i] = vx.m_int;
+		py[i] = vy.m_int;
+	}
+
+	surface()->DrawPolyLine( px, py, count );
+}
+
+void CScriptSurface::DrawOutlinedCircle( int x, int y, int radius, int segments )
+{
+	surface()->DrawOutlinedCircle( x, y, radius, segments );
+}
+
+void CScriptSurface::SetTextColor( int r, int g, int b, int a )
+{
+	surface()->DrawSetTextColor( r, g, b, a );
+}
+
+void CScriptSurface::SetTextPos( int x, int y )
+{
+	surface()->DrawSetTextPos( x, y );
+}
+
+void CScriptSurface::DrawText( const char *text, int drawType/* = FONT_DRAW_DEFAULT*/ )
+{
+	wchar_t wcs[512];
+	g_pVGuiLocalize->ConvertANSIToUnicode( text, wcs, sizeof(wcs) );
+	surface()->DrawPrintText( wcs, wcslen(wcs), (FontDrawType_t)drawType );
+}
+
+int CScriptSurface::GetFont( const char* name, bool proportional, const char* schema )
+{
+	HFont font = INVALID_FONT;
+
+	if ( !schema || !schema[0] )
+	{
+		int idx = g_ScriptFonts.Find( GetFixedFontName( name, proportional ) );
+		if ( idx != g_ScriptFonts.InvalidIndex() )
+		{
+			font = g_ScriptFonts[idx].font;
+		}
+	}
+	else
+	{
+		HScheme sch = scheme()->GetScheme( schema );
+		font = scheme()->GetIScheme(sch)->GetFont( name, proportional );
+
+		// Update known count
+		if ( font > (unsigned int)g_nFontCount )
+			g_nFontCount = font;
+	}
+
+	return HandleToInt( font );
+}
+
+void CScriptSurface::SetFont( int font )
+{
+	surface()->DrawSetTextFont( IntToFontHandle(font) );
+}
+
+int CScriptSurface::GetTextWidth( int font, const char* text )
+{
+	int w, t;
+	wchar_t wcs[512];
+	g_pVGuiLocalize->ConvertANSIToUnicode( text, wcs, sizeof(wcs) );
+	surface()->GetTextSize( IntToFontHandle(font), wcs, w, t );
+	return w;
+}
+
+int CScriptSurface::GetFontTall( int font )
+{
+	return surface()->GetFontTall( IntToFontHandle(font) );
+}
+
+int CScriptSurface::GetCharacterWidth( int font, int ch )
+{
+	return surface()->GetCharacterWidth( IntToFontHandle(font), ch );
+}
+
+int CScriptSurface::CreateFont( const char *customName, const char *windowsFontName, int tall, int weight, int blur, int scanlines, int flags, bool proportional )
+{
+	HFont font;
+	const char *fontAlias = GetFixedFontName( customName, proportional );
+
+	int idx = g_ScriptFonts.Find( fontAlias );
+	if ( idx != g_ScriptFonts.InvalidIndex() )
+	{
+		font = g_ScriptFonts[idx].font;
+
+		int oldTall = surface()->GetFontTall( font );
+		if ( proportional )
+			oldTall = scheme()->GetProportionalNormalizedValue( oldTall );
+
+		if ( oldTall == tall &&
+			!V_stricmp( surface()->GetFontName( font ), windowsFontName ) )
+		{
+			// Name and height match, this is most likely the same font.
+			return font;
+		}
+
+#ifdef _DEBUG
+		Msg( "CreateFont called on existing font [%i]%s\n", font, fontAlias );
+#endif
+	}
+	else
+	{
+		int size = V_strlen( windowsFontName ) + 1;
+		char *fontName = new char[ size ];
+		V_strncpy( fontName, windowsFontName, size );
```

Comment:
```suggestion
		char *fontName = strdup( windowsFontName );
```
Free via `free()` instead `delete[]`.

Some other places truncate the fontname to 64 bytes, maybe `FontData_t::name` could also become `char[64]` so no memory management is necessary.

Reply:
That 64 byte long string is the font alias allocated in the dictionary, not the font name.